include/logging/log_core.h

```diff
@@ -283,12 +320,51 @@ static inline char z_log_minimal_level_to_char(int level)
 		}							       \
 	} while (false)
 
+
+#define __LOG_VA(_level, _id, _filter, _format, _valist, _argnum, _strdup) \
+		do {									   \
+			bool is_user_context = _is_user_context();			   \
+											   \
+			if (Z_LOG_CONST_LEVEL_CHECK(_level)) {				   \
+				if (IS_ENABLED(CONFIG_LOG_MINIMAL)) {			   \
+					if (IS_ENABLED(CONFIG_LOG_PRINTK)) { \
+						log_printk(_format, _valist); \
+					} else { \
+						vprintk(_format, _valist); \
+					} \
+				} else if (is_user_context ||				   \
+					   (_level <= LOG_RUNTIME_FILTER(_filter))) {  \
+					struct log_msg_ids src_level = {		   \
+						.level = _level,			   \
+						.domain_id = CONFIG_LOG_DOMAIN_ID,	   \
+						.source_id = _id			   \
+					};						   \
+						__LOG_INTERNAL_VA(is_user_context, 	   \
+								   src_level,		   \
+								   _format, _valist, _argnum, _strdup); \
+				}							   \
+			}								   \
+			if (false) {							   \
```

Comment:
is this part needed?

Reply:
no needed if `log_printf_arg_checker` is not used